### PR TITLE
Add test for seeding algos

### PR DESCRIPTION
**Pull request by bouthilx**

### CODE REVIEW

File: tests/functional/algos/test_algos.py [hunk 1]
```diff
@@ -153,16 +152,24 @@ def xfail_if_not_installed(value: dict, import_optional: ImportOptional):
 
 
 class CustomRosenbrock(BenchmarkTask):
-    def __init__(self, max_trials: int = 30, with_fidelity: bool = False):
+    def __init__(
+        self, max_trials: int = 30, with_fidelity: bool = False, seed: int = None
```
Comment (lebrice) [inline tests/functional/algos/test_algos.py:156]:
```suggestion
        self, max_trials: int = 30, with_fidelity: bool = False, seed: int | None = None
```

File: tests/functional/algos/test_algos.py [hunk 2]
```diff
@@ -153,16 +152,24 @@ def xfail_if_not_installed(value: dict, import_optional: ImportOptional):
...
+    ):
         super().__init__(max_trials)
+        self.seed = seed
         self.with_fidelity = with_fidelity
+        self.initialize()
+
+    def initialize(self):
+        self.rng = numpy.random.RandomState(self.seed)
```
Comment (lebrice) [inline tests/functional/algos/test_algos.py:164]: Oh I see. You're using this method to re-initialize the RNG when reusing the task I think.

Nitpick: I don't remember if the `self.rng` attribute is already type-hinted in the base class. If it isn't, then it's better for the type checker (and also nicer IMO) to declare all attributes in `__init__`.
```suggestion
        self.rng: numpy.random.RandomState
        self.initialize()

    def initialize(self):
        self.rng = numpy.random.RandomState(self.seed)
```

File: tests/functional/algos/test_algos.py [hunk 3]
```diff
@@ -383,6 +391,41 @@ def test_with_fidelity(algorithm: dict):
...
+    ids=list(algorithm_configs.keys()),
+)
+def test_seeding(algorithm: dict):
+    """Test seeding works properly."""
+
+    rosenbrock_with_fidelity.initialize()
+    exp = workon(
+        rosenbrock_with_fidelity,
+        space_with_fidelity,
+        algorithms=algorithm,
```
Comment (lebrice) [inline tests/functional/algos/test_algos.py:406]: Is the PR for renaming "algorithms" -> "algorithm" in yet?
```suggestion
        algorithm=algorithm,
```

**Review by lebrice — APPROVED** [review]
Nice!

File: tests/functional/algos/test_algos.py (hunk 3 above)
Comment (bouthilx) [reply tests/functional/algos/test_algos.py:406]: Not yet but good catch. Let's harmonize already. Thanks!

File: tests/functional/algos/test_algos.py [hunk 4]
```diff
@@ -383,6 +391,41 @@ def test_with_fidelity(algorithm: dict):
...
+        max_trials=30,
+    )
+
+    assert exp.configuration["algorithms"] == algorithm
```
Comment (bouthilx) [inline tests/functional/algos/test_algos.py:410]:
```suggestion
    assert exp.configuration["algorithm"] == algorithm
```

File: tests/functional/algos/test_algos.py [hunk 5]
```diff
@@ -383,6 +391,41 @@ def test_with_fidelity(algorithm: dict):
...
+
+    rosenbrock_with_fidelity.initialize()
+    rep_exp = workon(
+        rosenbrock_with_fidelity,
+        space_with_fidelity,
+        algorithms=algorithm,
+        max_trials=30,
+    )
+
+    assert rep_exp.configuration["algorithms"] == algorithm
```
Comment (bouthilx) [inline tests/functional/algos/test_algos.py:422]:
```suggestion
    assert rep_exp.configuration["algorithm"] == algorithm
```

File: tests/functional/algos/test_algos.py [hunk 6]
```diff
@@ -383,6 +391,41 @@ def test_with_fidelity(algorithm: dict):
...
+
+    assert exp.configuration["algorithms"] == algorithm
+
+    trials = exp.fetch_trials()
+
+    rosenbrock_with_fidelity.initialize()
+    rep_exp = workon(
+        rosenbrock_with_fidelity,
+        space_with_fidelity,
+        algorithms=algorithm,
```
Comment (bouthilx) [inline tests/functional/algos/test_algos.py:418]:
```suggestion
        algorithm=algorithm,
```

**Comment by bouthilx** [comment]
Waiting for #1014 to be merged before merging this PR.

**Pull request merged**